src/main/java/domain/Referee.java

```diff
@@ -15,16 +24,80 @@ public static void decideWinner(Player player, Dealer dealer, Map<Gambler, Integ
         }
     }
 
-    private static boolean isPlayerWin(Player player, Dealer dealer) {
+    private boolean isPlayerWin(Player player, Dealer dealer) {
         int playerScore = player.getScore();
         int dealerScore = dealer.getScore();
         return (dealerScore <= playerScore && !player.isBustedGambler()
                 || (dealer.isBustedGambler() && !player.isBustedGambler()));
     }
 
-    private static boolean isDealerWin(Player player, Dealer dealer) {
+    private boolean isDealerWin(Player player, Dealer dealer) {
         int playerScore = player.getScore();
         int dealerScore = dealer.getScore();
         return dealerScore > playerScore || player.isBustedGambler();
     }
+
+    public void decideBenefits(Player player, List<Player> players, Map<Gambler, Integer> benefits) {
+        int index = players.indexOf(player);
+        int winnerCount = 0;
+
+        if (isPlayerWin(player, dealer) && !benefits.containsKey(player)) {
+            benefits.put(player, 0);
+        }
+
+        if (isDealerWin(player, dealer)) {
+            loseBenefit(player, benefits, bettings.get(index).getBetting());
+            winnerCount = getWinnerCount(player, players, winnerCount + 1);
+            int money = bettings.get(index).getBetting() / winnerCount;
+            earnBenefits(player, players, benefits, money);
+        }
+    }
+
+    private void earnBenefits(Player player, List<Player> players, Map<Gambler, Integer> benefits, int money) {
+        earnBenefit(benefits, money, dealer);
+        for (Player searchPlayer : players) {
+            playerEarnBenefit(player, benefits, money, searchPlayer);
+        }
+    }
+
+    private void playerEarnBenefit(Player player, Map<Gambler, Integer> benefits, int money, Player searchPlayer) {
+        if (searchPlayer != player && isPlayerWin(searchPlayer, dealer)) {
+            earnBenefit(benefits, money, searchPlayer);
+        }
+    }
+
+    private int getWinnerCount(Player player, List<Player> players, int winnerCount) {
+        for (Player searchPlayer : players) {
+            winnerCount = getWinnerCount(player, winnerCount, searchPlayer);
+        }
+        return winnerCount;
+    }
+
+    private int getWinnerCount(Player player, int winnerCount, Player searchPlayer) {
+        if (searchPlayer != player && isPlayerWin(searchPlayer, dealer)) {
+            return winnerCount + 1;
+        }
+        return winnerCount;
+    }
+
+    private void loseBenefit(Player player, Map<Gambler, Integer> benefits, int money) {
+        if (!benefits.containsKey(player)) {
+            benefits.put(player, 0 - money);
+            return;
+        }
+
+        if (benefits.containsKey(player)) {
+            benefits.replace(player, benefits.get(player) - money);
+        }
```

Comment:
이렇게 써도 되지 않을까요? (if문 긍정 및 else 사용)
```suggestion
        if (benefits.containsKey(player)) {
            benefits.replace(player, benefits.get(player) - money);
        } else {
            benefits.put(player, 0 - money);
        }
```

Reply:
빙봉, 코멘트 받고 많이 고민해보고 이렇게 수정해봤는데 어떻게 생각하세용?ㅎㅎ

1. benefits 처음 생성할 때 dealer, player를 넣고 초기값(0)을 설정해줬어요
2. 따라서 Referee class에서 containsKey() 코드를 모두 제거할 수 있었어요(당연히 있을 거니까)
3. method를 winDealer, winPlayer로 나눠 else를 쓰지 않고도 직관적으로 benefts를 구할 수 있다